Fastlane/testing_lanes.rb

```diff
@@ -56,18 +61,18 @@
       output_types: '',
       # xcodebuild_formatter: '', # Add this to get verbose logging by disabling xcbeautify.
       suppress_xcode_output: false,
-      buildlog_path: ENV['BITRISE_DEPLOY_DIR'],
+      buildlog_path: ENV['BITRISE_DEPLOY_DIR'], # By configuring `BITRISE_DEPLOY_DIR` we make sure our build log is deployed and available in Bitrise.
       prelaunch_simulator: true,
-      xcargs: "-clonedSourcePackagesDirPath #{sourcePackagesDir} -parallel-testing-enabled NO -retry-tests-on-failure -test-iterations 3",
+      xcargs: "-clonedSourcePackagesDirPath #{source_packages_dir} -parallel-testing-enabled NO -retry-tests-on-failure -test-iterations 3",
       include_simulator_logs: false, # Needed for this: https://github.com/fastlane/fastlane/issues/8909
       result_bundle: true,
       output_directory: "#{ENV['PWD']}/build/reports/",
-      derived_data_path: "#{ENV['PWD']}/build/derived_data", # Set buildlog and derived data path to fix permission issues on Bitrise
-      package_path: options[:package_path],
+      derived_data_path: "#{ENV['PWD']}/build/derived_data", # Set buildlog and derived data path to fix permission issues on Bitrise.
+      package_path: options[:package_path], # Optional path to the SPM package to test.
```

Comment:
Is this "to test this pull request" or is this used to pass package paths that we want to test? That feels odd, as I would expect that to be handled by Xcode. What does `scan` do again?

Reply:
It's used to pass package paths that we want to test. `scan` is Fastlane's tool around `xcodebuild`. You can use `xcodebuild` to test `.xcodeproj` files as well as packages. `package_path` is used to link to a package if required. We're passing in `nil` if we test an `xcodeproj`.

> That feels odd, as I would expect that to be handled by Xcode.

I assume you mean `xcodebuild` here. `xcodebuild` can't predict what you want to test. By default, it would search for an `xcodeproj` file, but it can't always find packages. Especially if these are located in subfolders which is the case when using our CI repository logic that's inside `Submodules/WeTransfer-iOS-CI`